File: 백준 1181 단어정렬 -정렬-/Main.java
import java.util.ArrayList;
import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int N = sc.nextInt();
        ArrayList<String> list = new ArrayList<>();
        for (int i = 0; i < N; i++) {
            String str = sc.next();
            if (!list.contains(str)) {
                list.add(str);
            }
        }
        list.sort((o1, o2) -> {
            if (o1.length() != o2.length()) {
                return o1.length() - o2.length();
            } else {
                return o1.compareTo(o2);
            }
        });
        for (String str : list) {
            System.out.println(str);
        }
    }
}

File: 백준 1181 단어정렬 -정렬-/Main.kt
import java.util.*

fun main(args: Array<String>) {
    val sc = Scanner(System.`in`)
    val N = sc.nextInt()
    val list = ArrayList<String>()
    for (i in 0 until N) {
        val str = sc.next()
        if (!list.contains(str)) {
            list.add(str)
        }
    }
    list.sortWith(Comparator sort@{ o1: String, o2: String ->
        if (o1.length != o2.length) {
            return@sort o1.length - o2.length
        } else {
            return@sort o1.compareTo(o2)
        }
    })
    for (str in list) {
        println(str)
    }
}